### PR TITLE
Store Submission State 

### PR DESCRIPTION
**Pull request by orta**
Blocked by #756

Uses the react-native [AsyncStorage](https://facebook.github.io/react-native/docs/asyncstorage.html) key value store to store and restore a user's consignment session.

Touches on https://github.com/artsy/consignments/issues/70 but fixes https://github.com/artsy/consignments/issues/84

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,78 +1,67 @@
...
+  "files.exclude": {
+    "**/.git": true,
+    "**/.svn": true,
+    "**/.hg": true,
+    "**/.DS_Store": true,
+    ".jest/": true,
+    "build": true,
+    ".awcache": true,
+    ".happypack": true,
+    "Example/build": true,
```
Comment (orta) [inline .vscode/settings.json]:
```diff
+    ".awcache": true,
+    ".happypack": true,
```

is the real change here, but I removed the comments so prettier can handle the JSON formatting

File: src/lib/Components/Consignments/Screens/Overview.tsx [hunk 2]
```diff
@@ -123,7 +139,7 @@ export default class Info extends React.Component<Props, ConsignmentSetup> {
 
             <Row style={{ justifyContent: "center" }}>
               <View style={{ height: 43, width: 320, marginTop: 20, opacity: canSubmit ? 1 : 0.3 }}>
-                <Button text="NEXT" onPress={canSubmit && this.goToFinalSubmission} style={{ flex: 1 }} />
+                <Button text="NEXT" onPress={canSubmit ? this.goToFinalSubmission : emptyFunc} style={{ flex: 1 }} />
```
Comment (orta) [inline src/lib/Components/Consignments/Screens/Overview.tsx]: This was causing runtime warnings

File: src/lib/Components/Consignments/__stories__/Consignments.story.tsx [hunk 3]
```diff
@@ -31,7 +31,8 @@ const blankMetadata = {
 storiesOf("Consignments/_Screens")
   .add("Consignments", () => <Nav />)
   .add("Welcome Page", () => <Welcome navigator={nav} route={route} />)
-  .add("Overview Page", () => <Overview navigator={nav} route={route} setup={{}} />)
+  .add("Overview Page (Fresh)", () => <Overview navigator={nav} route={route} setup={{}} />)
+  .add("Overview Page", () => <Overview navigator={nav} route={route} setup={null} />)
```
Comment (orta) [inline src/lib/Components/Consignments/__stories__/Consignments.story.tsx]: This one will always be blank, and the other will always try restore the last version

File: src/lib/Components/Consignments/Screens/Overview.tsx [hunk 4]
```diff
@@ -67,9 +79,11 @@ export default class Info extends React.Component<Props, ConsignmentSetup> {
   updateLocation = (city: string, state: string, country: string) =>
     this.updateStateAndMetaphysics({ location: { city, state, country } })
 
-  updateStateAndMetaphysics = (state: any) => this.setState(state, this.updateMetaphysics)
+  updateStateAndMetaphysics = (state: any) => this.setState(state, this.updateLocalStateAndMetaphysics)
+
+  updateLocalStateAndMetaphysics = async () => {
+    this.saveStateToLocalStorage()
```
Comment (xtina-starr) [inline src/lib/Components/Consignments/Screens/Overview.tsx]: The method says it updates State and Metaphysics, but it seems it just set the item on the `AsyncStorage` (update the local store). I'm mainly curious if it should also be updating metaphysics (or if it does some where and I missed that). If it is indeed just updating the local store do it does it need to be called in the `submitFinalSubmission` on like 97?

Comment (orta) [reply src/lib/Components/Consignments/Screens/Overview.tsx]: The lines underneath ( 73-77 ) update Metaphysics (`updateSubmission` and `createSubmission`) - they're still around 👍

Comment (xtina-starr) [reply src/lib/Components/Consignments/Screens/Overview.tsx]: gotcha!

**Comment by xtina-starr** [comment]
👍

**Pull request merged**